libs/s25main/buildings/BurnedWarehouse.cpp

```diff
@@ -75,57 +73,49 @@ void BurnedWarehouse::HandleEvent(const unsigned /*id*/)
 
     for(const auto job : helpers::enumRange<Job>())
     {
-        // Anzahl ausrechnen, die in dieser Runde rausgeht
+        // In the last phase all remaining ones leave, else only some
         unsigned count;
         if(go_out_phase + 1 >= GO_OUT_PHASES)
-            count = people[job]; // Take all on last round
+            count = people[job];
         else
             count = people[job] / (GO_OUT_PHASES - go_out_phase);
+        if(count == 0)
+            continue;
 
-        // Von der vorhandenen Abzahl abziehen
+        // Remove from inventory
         people[job] -= count;
 
-        // In Alle Richtungen verteilen
-        // Startrichtung zufällig bestimmen
-        unsigned start_dir = RANDOM_RAND(helpers::NumEnumValues_v<Direction>);
+        // Distribute in all directions starting at a random one of the possible ones
+        const unsigned startIdx = (possibleDirs.size() <= 1u) ? 0 : RANDOM_RAND(possibleDirs.size());
+        const unsigned numPeoplePerDir = count / possibleDirs.size();
 
-        for(unsigned j = 0; j < possibleDirCt; ++j)
+        for(const unsigned j : helpers::range(possibleDirs.size()))
         {
-            // Aktuelle Richtung, die jetzt dran ist bestimmen
-            Direction dir = possibleDirs[j] + start_dir;
-
-            // Anzahl jetzt für diese Richtung ausrechnen
-            unsigned numPeopleInDir = count / possibleDirCt;
-            // Bei letzter Richtung noch den übriggebliebenen Rest dazuaddieren
-            if(j + 1 == possibleDirCt)
-                numPeopleInDir += count % possibleDirCt;
-
-            // Die Figuren schließlich rausschicken
-            for(unsigned z = 0; z < numPeopleInDir; ++z)
+            // Get current direction accounting for startIdx and hence possible wrap around
+            const unsigned idx = j + startIdx;
+            const Direction curDir = possibleDirs[idx < possibleDirs.size() ? idx : idx - possibleDirs.size()];
+            // Take all in last direction
+            const auto curNumPeople = (j + 1u < possibleDirs.size()) ? numPeoplePerDir : count;
+            count -= curNumPeople;
+            for(const auto z : helpers::range(curNumPeople))
```

Comment:
Should have tested with GCC, sorry for that -->
```suggestion
            for([[maybe_unused]] const auto z : helpers::range(curNumPeople))
```